src/analysis/special_functions/trigonometric/angle.lean

```diff
@@ -521,6 +521,23 @@ lemma eq_iff_abs_to_real_eq_of_sign_eq {θ ψ : angle} (h : θ.sign = ψ.sign) :
   θ = ψ ↔ |θ.to_real| = |ψ.to_real| :=
 by simpa [h] using @eq_iff_sign_eq_and_abs_to_real_eq θ ψ
 
+lemma continuous_at_sign {θ : angle} (h0 : θ ≠ 0) (hpi : θ ≠ π) : continuous_at sign θ :=
+(continuous_at_sign_of_ne_zero (sin_ne_zero_iff.2 ⟨h0, hpi⟩)).comp continuous_sin.continuous_at
+
+/-- Suppose a function to angles is continuous on a connected set and never takes the values `0`
+or `π` on that set. Then the values of the function on that set all have the same sign. -/
+lemma sign_eq_of_continuous_on {α : Type*} [topological_space α] {f : α → angle} {s : set α}
+  {x y : α} (hc : is_connected s) (hf : continuous_on f s) (hs : ∀ z ∈ s, f z ≠ 0 ∧ f z ≠ π)
+  (hx : x ∈ s) (hy : y ∈ s) : (f y).sign = (f x).sign :=
+begin
+  have hf' : continuous_on (sign ∘ f) s,
+  { refine (continuous_at.continuous_on (λ θ hθ, _)).comp hf (set.maps_to_image f s),
+    obtain ⟨z, hz, rfl⟩ := hθ,
+    exact continuous_at_sign (hs _ hz).1 (hs _ hz).2 },
```

Comment:
Wouldn't that make a good lemma on its own?

Reply:
That specific statement about signs of angles, or some more general composition of these topological lemmas (there are endless ways in which basic topological lemmas could be composed)? I don't currently have an anticipated use for that statement outside of this particular lemma.

Reply:
I think that in general it is nice to have several versions of a continuity lemma and since you need the `continuous_on` version already, it should not hurt to add it as a standalone lemma

Reply:
I've now split out that lemma.